### PR TITLE
[x264] Update to 0.164.3108, use official upstream

### PR DESCRIPTION
**Pull request by dg0yt**
Using the last commit in branch stable.

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/x264/configure.patch [hunk 1]
```diff
@@ -0,0 +1,40 @@
+diff --git a/configure b/configure
+index e242e73c..e0d1df76 100755
+--- a/configure
++++ b/configure
+@@ -1,4 +1,7 @@
+-#!/bin/bash
++/usr/bin/env bash
```
Comment (BillyONeal) [inline ports/x264/configure.patch:7]:
```suggestion
+#!/usr/bin/env bash
```

Doesn't it still need to be a shebang?

**Review by BillyONeal — COMMENTED** [review]
This LGTM except for the shebang thing.... about which I admit I am extremely underinformed...

File: ports/x264/configure.patch (hunk 1 above)
Comment (dg0yt) [reply ports/x264/configure.patch:7]: This is not a addition but ported from `vcpkg_replace_string` (#26563) into the patch. I do not know all the macOS linux bsd qnx whatever particularities.

Comment (dg0yt) [reply ports/x264/configure.patch]: :facepalm: Me needs more sleep. Thanks for catching.

Comment (BillyONeal) [reply ports/x264/configure.patch]: No problem! :)

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**